sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -371,6 +390,10 @@ func (c *control) handleInstruction(ctx context.Context, req *fnpb.InstructionRe
 		c.cache.CompleteBundle(tokens...)
 
 		mons, pylds := monitoring(plan, store)
+		if c.runnerCapabilities[URNMonitoringInfoShortID] {
+			mons = []*pipeline_v1.MonitoringInfo{}
+		}
```

Comment:
So this only accomplishes half the goal of the short-ids approach to monitoring. While we are no longer sending the larger payloads, we are still allocating and authoring them, which we should still avoid doing if we can avoid it. The monitoring method is what needs changing to "understand" the capabilities and provide one thing or another (or if we have no information, both).

Reply:
yeah, that is taking up memory as well as space. Fixed.